### PR TITLE
Reading encryption in case of api key reset

### PR DESCRIPTION
**Pull request by GunaKKIBM**
Changes as a part of this PR:

- Fix for a customer issue - https://github.ibm.com/alchemy-containers/customer-tickets/issues/12440
- Improved secret watcher in case of secret update.
- Golang version update to 1.18.2

### CODE REVIEW

File: pkg/ibmcsidriver/node.go [hunk 1]
```diff
@@ -285,7 +286,8 @@ func (csiNS *CSINodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeSt
 	// This operation (NodeStageVolume) MUST be idempotent.
 	// If the volume corresponding to the volume_id is already staged to the staging_target_path,
 	// and is identical to the specified volume_capability the Plugin MUST reply 0 OK.
-	if device == source {
+	target, err := filepath.EvalSymlinks(source)
+	if err == nil && device == target {
```
Comment (prankulmahajan) [inline pkg/ibmcsidriver/node.go:290]: Can we handle err != nil ?

Comment (GunaKKIBM) [reply pkg/ibmcsidriver/node.go:290]: If err != nil, there was some issue, in finding the softlink that source. But, that is not a blocker for us to go ahead and mount the volume. All we can do is log the error. But, just for the sake of logging the error we have to add an else condition.

Comment (prankulmahajan) [reply pkg/ibmcsidriver/node.go:290]: Got it!

**Review by prankulmahajan — COMMENTED** [review]
/lgtm

**Review by sameshai — APPROVED** [review]
/lgtm

**Review by prankulmahajan — COMMENTED** [review]
/lgtm

**Review by sameshai — APPROVED** [review]
/lgtm

**Comment by GunaKKIBM** [comment]
Changes as a part of this PR:
- Fix for format and mount failed issue - In case of retry to stage the volume from the kubelet, if the volume is already staged, driver returns with success response. 
- Reading encryption variable also in case of api key reset
- Golang version update to 1.18.2

**Review by arahamad — APPROVED** [review]
/lgtm

**Pull request merged**